toolchain/cc_toolchain_config.bzl

```diff
@@ -0,0 +1,151 @@
+load("@bazel_tools//tools/build_defs/cc:action_names.bzl", "ACTION_NAMES")
+load(
+    "@bazel_tools//tools/cpp:cc_toolchain_config_lib.bzl",
+    "feature",
+    "flag_group",
+    "flag_set",
+    "tool",
+    "tool_path",
+)
+
+all_link_actions = [
+    ACTION_NAMES.cpp_link_executable,
+    ACTION_NAMES.cpp_link_dynamic_library,
+    ACTION_NAMES.cpp_link_nodeps_dynamic_library,
+]
+
+all_compile_actions = [
+    ACTION_NAMES.assemble,
+    ACTION_NAMES.preprocess_assemble,
+    ACTION_NAMES.linkstamp_compile,
+    ACTION_NAMES.c_compile,
+    ACTION_NAMES.cpp_compile,
+    ACTION_NAMES.cpp_header_parsing,
+    ACTION_NAMES.cpp_module_codegen,
+    ACTION_NAMES.cpp_module_compile,
+    ACTION_NAMES.clif_match,
+    ACTION_NAMES.lto_backend,
+]
+
+def _impl(ctx):
+  tool_paths = [
+      tool_path(
+          name = "gcc",
+          path = "/usr/bin/clang",
+      ),
+      tool_path(
+          name = "ld",
+          path = "/usr/bin/ld",
+      ),
+      tool_path(
+          name = "ar",
+          path = "/usr/bin/llvm-ar-11",
+      ),
+      tool_path(
+          name = "compat-ld",
+          path = "/usr/bin/ld",
+      ),
+      tool_path(
+          name = "cpp",
+          path = "/usr/bin/clang-cpp",
+      ),
+      tool_path(
+          name = "dwp",
+          path = "/usr/bin/llvm-dwp",
+      ),
+      tool_path(
+          name = "gcov",
+          path = "/usr/bin/llvm-profdata",
+      ),
+      tool_path(
+          name = "nm",
+          path = "/usr/bin/llvm-nm",
+      ),
+      tool_path(
+          name = "objcopy",
+          path = "/usr/bin/llvm-objcopy",
+      ),
+      tool_path(
+          name = "objdump",
+          path = "/usr/bin/llvm-objdump",
+      ),
+      tool_path(
+          name = "strip",
+          path = "/usr/bin/llvm-strip",
+      ),
+  ]
+
+  features = [
+      feature(
+          name = "default_linker_flags",
+          enabled = True,
+          flag_sets = [
+              flag_set(
+                  actions = all_link_actions,
+                  flag_groups = ([
+                      flag_group(
+                          flags = [
+                              "-lstdc++",
+                              "--target=" + ctx.attr.target_full_name,
+                          ],
+                      ),
+                  ]),
+              ),
+          ],
+      ),
+  ]
+
+  features.append(feature(
```

Comment:
any reason to not just inline this as part of the array literal? vs calling append?

Reply:
Nope, added it to part of the array literal.